### ISSUE TITLE
`getPluginRows` postgres query needs optimization

### ISSUE THREAD

**Issue by macobo** [PostHog/posthog#10308]
## Bug description

Heroku is reporting that the following query is super slow:

![image](https://user-images.githubusercontent.com/148820/173813665-97094adb-5f97-4b2d-9156-1e9e5b8f26bc.png)

We should fix this

## Environment

- [x] PostHog Cloud
- [ ] self-hosted PostHog (ClickHouse-based), version/commit: _please provide_
- [ ] self-hosted PostHog (Postgres-based, legacy), version/commit: _please provide_

## Additional context

cc @Twixes interested in grabbing this?

**Comment by Twixes** [comment]
I digged into this a bit and the main culprit seems to be… IO: the `getPluginRows()` query returns ~45 MB of data in the form of the `posthog_plugin.archive` blob column. With `archive` excluded, it completes in a reasonable 200-300 ms, but `*` includes `archive`, balooning that. One way to improve this would be to save all files needed to run a plugin into `posthog_pluginsourcefile` at plugin installation time – there are only three we need: `plugin.json`, `index.ts` (or an alias defined in `plugin.json`'s `main`) and now `frontend.tsx`. Most of archive contents is junk at runtime, if we only loaded the required code this wouldn't be an issue (in the medium term at least).
